pkg/install/fixpullsecret.go

```diff
@@ -0,0 +1,37 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/util/pullsecret"
+)
+
+func (i *Installer) fixPullSecret(ctx context.Context) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		ps, err := i.kubernetescli.CoreV1().Secrets("openshift-config").Get("pull-secret", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		dockerconfig := string(ps.Data[".dockerconfigjson"])
+		for _, rp := range i.doc.OpenShiftCluster.Properties.RegistryProfiles {
+			if strings.Contains(rp.Name, ".azurecr.io") {
```

Comment:
I think this needs more thinking not to step over on customer actions. `arosvc.azurecr.io` will check it.